### PR TITLE
Upgrade protobuf to 3.17.0

### PR DESCRIPTION
**Pull request by tandel-pratik**
* Some of these artifacts still need to be uploaded to the bazel mirrors it seems and i'm not sure i have permissions to do that.

### CODE REVIEW

**Comment by tandel-pratik** [comment]
hey @comius @lberki is it possible to merge this? these proto upgrade contains some new kotlin generated bindings that we'd find really useful

**Review by Yannic — APPROVED** [review]
LGTM

@meteorcloudy Can you please help with uploading the following files?
`https://github.com/protocolbuffers/protobuf/archive/v3.17.0.tar.gz`
`https://github.com/protocolbuffers/protobuf/releases/download/v3.17.0/protoc-3.17.0-linux-aarch_64.zip`
`https://github.com/protocolbuffers/protobuf/releases/download/v3.17.0/protoc-3.17.0-linux-ppcle_64.zip`
`https://github.com/protocolbuffers/protobuf/releases/download/v3.17.0/protoc-3.17.0-linux-s390_64.zip`
`https://github.com/protocolbuffers/protobuf/releases/download/v3.17.0/protoc-3.17.0-linux-x86_32.zip`
`https://github.com/protocolbuffers/protobuf/releases/download/v3.17.0/protoc-3.17.0-linux-x86_64.zip`
`https://github.com/protocolbuffers/protobuf/releases/download/v3.17.0/protoc-3.17.0-osx-x86_64.zip`
`https://github.com/protocolbuffers/protobuf/releases/download/v3.17.0/protoc-3.17.0-win32.zip`
`https://github.com/protocolbuffers/protobuf/releases/download/v3.17.0/protoc-3.17.0-win64.zip`
`https://repo1.maven.org/maven2/com/google/protobuf/protobuf-java/3.17.0/protobuf-java-3.17.0.jar`
`https://repo1.maven.org/maven2/com/google/protobuf/protobuf-java/3.17.0/protobuf-java-3.17.0-sources.jar`
`https://repo1.maven.org/maven2/com/google/protobuf/protobuf-java-util/3.17.0/protobuf-java-util-3.17.0.jar`
`https://repo1.maven.org/maven2/com/google/protobuf/protobuf-java-util/3.17.0/protobuf-java-util-3.17.0-sources.jar`
`https://repo1.maven.org/maven2/com/google/protobuf/protobuf-javalite/3.17.0/protobuf-javalite-3.17.0.jar`
`https://repo1.maven.org/maven2/com/google/protobuf/protobuf-javalite/3.17.0/protobuf-javalite-3.17.0-sources.jar`

**Comment by meteorcloudy** [comment]
@Yannic The artifacts should be available through Bazel mirror.

**Comment by tandel-pratik** [comment]
hey @Yannic @meteorcloudy can this be merged now that the artifacts are available?

File: proto/private/dependencies.bzl [hunk 1]
```diff
@@ -23,75 +23,75 @@ dependencies = {
...
+        "sha256": "1c4adfbf4ba67a16353dd5f4ebaa9abb998684f9c0fbdccea7313e256afcc65e",
         "urls": [
-            "https://mirror.bazel.build/github.com/protocolbuffers/protobuf/releases/download/v3.13.0/protoc-3.13.0-linux-ppcle_64.zip",
-            "https://github.com/protocolbuffers/protobuf/releases/download/v3.13.0/protoc-3.13.0-linux-ppcle_64.zip",
+            "https://mirror.bazel.build/github.com/protocolbuffers/protobuf/releases/download/v3.17.0/protoc-3.17.0-linux-ppcle_64.zip",
+            "https://github.com/protocolbuffers/protobuf/releases/download/v3.17.0/protoc-3.17.0-linux-ppcle_64.zip",
         ],
     },
-    "com_google_protobuf_protoc_linux_s390x": {
+    "com_google_protobuf_protoc_linux_s390_64": {
```
Comment (comius) [inline proto/private/dependencies.bzl:49]: Please fix reference to com_google_protobuf_protoc_linux_s390 in proto/private/BUILD.release.

**Review by comius — CHANGES_REQUESTED** [review]

**Review by comius — APPROVED** [review]

**Comment by kconvey** [comment]
@meteorcloudy What is the URL for the bazel mirror? Can we update `README.md` to reflect supporting 3.17.0? 

Happy to do it if I have the new URL.

**Comment by meteorcloudy** [comment]
See https://github.com/bazelbuild/rules_proto/pull/88/files#diff-374db506507a0693f52d91212c45c56983f9d38960cd9ad732fcf774b68d201bR29

What README.md do you mean?

**Pull request merged**